lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyLink.java

```diff
@@ -77,9 +96,32 @@ public NettyLink(final Channel channel, final Encoder<? super T> encoder, final
   @Override
   public void write(final T message) {
     LOG.log(Level.FINEST, "write {0} :: {1}", new Object[] {channel, message});
-    final ChannelFuture future = channel.writeAndFlush(Unpooled.wrappedBuffer(encoder.encode(message)));
-    if (listener !=  null) {
-      future.addListener(new NettyChannelFutureListener<>(message, listener));
+
+    if (this.uri != null) {
```

Comment:
I think creating a `NettyHttpLink` would be better instead of fixing this class. 
We have to check whenever `.write` is called whether the uri is null or not. This doesn't look good to me.